diff --git a/libavcodec/aac.c b/libavcodec/aac.c
index e8c2f65..b2fd6f9 100644
--- a/libavcodec/aac.c
+++ b/libavcodec/aac.c
@@ -274,7 +274,9 @@ static int decode_pce(AACContext *ac, enum ChannelPosition new_che_pos[4][MAX_EL
 {
     int num_front, num_side, num_back, num_lfe, num_assoc_data, num_cc, sampling_index;
 
-    skip_bits(gb, 2);  // object_type
+    int aot = get_bits(gb, 2) + 1;  // object_type
+    if (ac->m4ac.object_type != aot)
+        av_log(ac->avccontext, AV_LOG_WARNING, "AOT in the program config element (%d) does not match the AOT in the m4ac (%d).\n", aot, ac->m4ac.object_type);
 
     sampling_index = get_bits(gb, 4);
     if (ac->m4ac.sampling_index != sampling_index)
@@ -497,6 +499,10 @@ static av_cold int aac_decode_init(AVCodecContext *avccontext)
     ac->avccontext = avccontext;
 
     if (avccontext->extradata_size > 0) {
+        av_log(avccontext, AV_LOG_ERROR, "extradata[%d]: ", avccontext->extradata_size);
+        for (i = 0; i < avccontext->extradata_size; i++)
+            av_log(avccontext, AV_LOG_ERROR, "%02x ", (int)((unsigned char*)avccontext->extradata)[i]);
+        av_log(avccontext, AV_LOG_ERROR, "\n");
         if (decode_audio_specific_config(ac, avccontext->extradata, avccontext->extradata_size))
             return -1;
         avccontext->sample_rate = ac->m4ac.sample_rate;
